build/dsc/psmodule.rb

```diff
@@ -18,7 +18,7 @@ def attributes
       begin
         unless @attributes
           attrs = {}
-          regex = /^(.*) *= *['"](.*)['"] *$/
+          regex = /^(.*) *= *['"](.*)['";] *$/
```

Comment:
This regex now allows `;` to terminate a string like this:

```
[2] pry(main)> "foo = 'bar;" =~ /^(.*) *= *['"](.*)['";] *$/
=> 0
```

Even though this regex doesn't do the balanced quote matching properly, we probably want something that captures an optional `;` in the right spot like `/^(.*) *= *['"](.*)['"] *(;)? *$/` (normally I'd use `\s` for whitespace, but since this regex is already using a literal `<space>` I left that bit as-is)


```
[3] pry(main)> "foo = 'bar;" =~ /^(.*) *= *['"](.*)['"] *(;)? *$/
=> nil

[4] pry(main)> "foo = 'bar';" =~ /^(.*) *= *['"](.*)['"] *(;)? *$/
=> 0

[5] pry(main)> "foo = 'bar'" =~ /^(.*) *= *['"](.*)['"] *(;)? *$/
=> 0

[6] pry(main)> "foo = \"bar\"" =~ /^(.*) *= *['"](.*)['"] *(;)? *$/
=> 0
```

Reply:
Since you already went through the extra work I'll add it, but I don't think it's worth our time to be defensive against improper syntax. The format you suggest wouldn't have passed Test-ModuleManifest or the gallery. I don't want to be in the business of re-implementing the PowerShell tokenizer, as using regex is on the path to [evil](https://stackoverflow.com/questions/1732348/regex-match-open-tags-except-xhtml-self-contained-tags)

Reply:
Balanced HTML tags is a much much more difficult task than balancing quotes - don't fear the regex! ;0

I should have also pointed out your mod only matched because of the `(.*)` in the middle (which really should be `(\w*)`) greedily includes the trailing `'`, which means it returns the wrong captures and then sets the wrong value at line 28:

```
### original suggestion - notice capture 2 has a trailing ' after bar
[4] pry(main)> matches = "foo = 'bar';".match(/^(.*) *= *['"](.*)['";] *$/)
=> #<MatchData "foo = 'bar';" 1:"foo " 2:"bar'">

### new version - capture 2 doesn't have the trailing ' after bar
[5] pry(main)> matches = "foo = 'bar';".match(/^(.*) *= *['"](.*)['"] *(;)? *$/)
=> #<MatchData "foo = 'bar';" 1:"foo " 2:"bar" 3:";">
```

So this needed to be changed for that reason